src/main/java/com/yelp/nrtsearch/server/luceneserver/SearchHandler.java

```diff
@@ -191,7 +191,8 @@ public SearchResponse handle(IndexState indexState, SearchRequest searchRequest)
 
       diagnostics.setFirstPassSearchTimeMs(((System.nanoTime() - searchStartTime) / 1000000.0));
 
-      DeadlineUtils.checkDeadline("SearchHandler: post recall", "SEARCH");
+      DeadlineUtils.checkDeadline(
```

Comment:
Maybe add a diagnostics parameter to `checkDeadline` function, and append it to the message only when the error is being thrown? You can also consider keeping the existing function and adding a new checkDeadline with additional parameter in the signature.

Reply:
The method is getting used in many places in the code, so I will create a new method with the additional param.